### PR TITLE
feat: patch and delete

### PR DESCRIPTION
**Pull request by justyna-13**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -134,6 +134,130 @@ app.post("/api/users", async (req, res) => {
   res.json(user);
 });
 
+/**
+ * @openapi
+ * /api/users/{id}:
+ *   patch:
+ *      summary: Update particular data for a user
+ *      description: Ur
```
Comment (FRSgit) [inline src/index.ts]: typo I guess 😄

File: src/index.ts [hunk 2]
```diff
@@ -134,6 +134,130 @@ app.post("/api/users", async (req, res) => {
...
+ *      parameters:
+ *        - name: id
+ *          in: path
```
Comment (FRSgit) [inline src/index.ts:145]: Good usage of path parameter `type`!

File: src/index.ts [hunk 3]
```diff
@@ -134,6 +134,130 @@ app.post("/api/users", async (req, res) => {
...
+ *                          type: integer
+ *                          description: User id
+ *                          example: 1
+ *                        name:
+ *                          type: string
+ *                          description: Username
+ *                        email:
+ *                          type: string
+ *                          description: Username
+ *        400:
```
Comment (FRSgit) [inline src/index.ts:188]: Missing documentation for `404` response when user is missing

File: src/index.ts [hunk 4]
```diff
@@ -134,6 +134,130 @@ app.post("/api/users", async (req, res) => {
...
+  }
+
+  try {
+    const user = await prisma.user.update({
+      where: {
+        id: Number(id)
+      },
+      data: {
+        name: data.name || undefined,
+        email: data.email || undefined,
```
Comment (FRSgit) [inline src/index.ts:211]: Nice one! As [prisma guide states](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined) `undefined` means `do nothing`

File: src/index.ts [hunk 5]
```diff
@@ -134,6 +134,130 @@ app.post("/api/users", async (req, res) => {
...
+      },
+      data: {
+        name: data.name || undefined,
+        email: data.email || undefined, 
+      }
+    })
+
+    res.json(user);
+  } catch (e) {
+    res.status(404).json({});
```
Comment (FRSgit) [inline src/index.ts]: That's enough for our example, but just so you know - there are different error types that can appear here and with your code I'm able to get to this catch block not only by specifying incorrect ID, but also when I send e.g. 
```json
{
  "email": 1
}
```

as the body. I'll just get [PrismaClientValidationError](https://www.prisma.io/docs/reference/api-reference/error-reference#prismaclientvalidationerror) that says that `email` should be a string and not an integer. But don't worry, we'll cover this case when working on prisma in details later on - this is a good start!

File: src/index.ts [hunk 6]
```diff
@@ -134,6 +134,130 @@ app.post("/api/users", async (req, res) => {
...
+  }
+
+  try {
+    const user = await prisma.user.delete({
+      where: {
+        id: Number(id),
+      },
+    });
+
+    res.json(user);
```
Comment (FRSgit) [inline src/index.ts:259]: Prisma makes is easy to implement proper user `DELETE` endpoint, doesn't it? 😄

File: src/index.ts [hunk 7]
```diff
@@ -134,6 +134,130 @@ app.post("/api/users", async (req, res) => {
...
+ *      description: delete a pet
+ *      parameters:
+ *        - name: id
+ *          in: path
+ *          description: User id to delete
+ *          required: true
+ *          schema:
+ *            type: integer
+ *      responses:
+ *        400:
```
Comment (FRSgit) [inline src/index.ts:237]: missing description of 200 response

**Review by FRSgit — COMMENTED** [review]
Great implementation - only two missing descriptions, but that's nothing to worry about - good job!

File: src/index.ts (hunk 1 above)
Comment (justyna-13) [reply src/index.ts]: right:) fixed

File: src/index.ts (hunk 3 above)
Comment (justyna-13) [reply src/index.ts:188]: fixed

File: src/index.ts (hunk 6 above)
Comment (justyna-13) [reply src/index.ts:259]: yes, I used it above and I just wanted to send info about deleted user (like a confirmation), is it ok?

File: src/index.ts (hunk 7 above)
Comment (justyna-13) [reply src/index.ts:237]: fixed

File: src/index.ts (hunk 6 above)
Comment (FRSgit) [reply src/index.ts:259]: yes, it does make sense - in many REST apis that's exactly how it works 🙂
As a side note - It's good to remember that sometimes there are some sensitive information in user data, so we need to be careful to not send back something we shouldn't 😄

**Comment by FRSgit** [comment]
Nicely done! I'm closing this PR as everything was done, thank you!

**Pull request closed without merging**